src/SimpleEventStore/SimpleEventStore.CosmosDb/AzureCosmosDbStorageEngine.cs

```diff
@@ -124,14 +126,31 @@ FROM e
 
         private Task<DatabaseResponse> CreateDatabaseIfItDoesNotExist()
         {
-            return _client.CreateDatabaseIfNotExistsAsync(_databaseName, _databaseOptions.DatabaseRequestUnits);
+            if (_databaseOptions.UseAutoscale)
+            {
+                if (_databaseOptions.DatabaseRequestUnits != null)
+                {
+                    return _client.CreateDatabaseIfNotExistsAsync(_databaseName,
+                        ThroughputProperties.CreateAutoscaleThroughput((int)_databaseOptions.DatabaseRequestUnits));
+                }
+                else
+                {
+                    return _client.CreateDatabaseIfNotExistsAsync(_databaseName,
```

Comment:
If we removed the default offer support then we could remove the logic in the else block